src/main/java/fi/aalto/cs/apluscourses/presentation/MainViewModel.java

```diff
@@ -125,4 +134,15 @@ public Options getExerciseFilterOptions() {
     return exerciseFilterOptions;
   }
 
+  private void setInGrading(List<ExerciseGroup> exerciseGroups) {
+    Map<Long, Exercise> exercises = new LinkedHashMap<>();
+    exerciseGroups.stream().forEach(group -> exercises.putAll(group.getExercises()));
+    if (exercises.containsKey(submittedForGrading.getId())) {
+      exercises.get(submittedForGrading.getId()).setInGrading(true);
+    }
```

Comment:
perhaps an improvement to the functional approach:
```java
final long id = submittedForGrading.getId();

    exerciseGroups
            .stream()
            .map(ExerciseGroup::getExercises)
            .filter(exercise -> exercise.containsKey(id))
            .forEach(exercise -> exercise.get(id).setInGrading(true));
```
which is equivalent to
```java
final long id = submittedForGrading.getId();

    for (ExerciseGroup exerciseGroup : exerciseGroups) {
      Map<Long, Exercise> exercise = exerciseGroup.getExercises();
      if (exercise.containsKey(id)) {
        exercise.get(id).setInGrading(true);
      }
    }
```
?